kv/kvnemesis: TestKVNemesisSingleNode failed

kv/kvnemesis.TestKVNemesisSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13821796?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13821796?buildTab=artifacts#/) on release-22.2 @ [3e2fa482345754749d3a51a612f7a482699ca1b3](https://github.com/cockroachdb/cockroach/commits/3e2fa482345754749d3a51a612f7a482699ca1b3):


```
=== RUN   TestKVNemesisSingleNode
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestKVNemesisSingleNode3580340144
    test_log_scope.go:79: use -show-logs to present logs inline
    kvnemesis_test.go:68: failure:
        error applying x.Barrier(ctx, /Table/100/"47ad2f1e", /Table/100/"4a2b6666") // result is ambiguous: replica removed: result is ambiguous: replica removed
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:894
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:444
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:69
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis.go:97
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisSingleNode
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:64
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1446
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.Barrier(ctx, /Table/100/"47ad2f1e", /Table/100/"4a2b6666") // result is ambiguous: replica removed
        Wraps: (3) result is ambiguous: replica removed
        Wraps: (4)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.init
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:61
          | runtime.doInit
          | 	GOROOT/src/runtime/proc.go:6348
          | runtime.doInit
          | 	GOROOT/src/runtime/proc.go:6325
          | runtime.doInit
          | 	GOROOT/src/runtime/proc.go:6325
          | runtime.doInit
          | 	GOROOT/src/runtime/proc.go:6325
          | runtime.main
          | 	GOROOT/src/runtime/proc.go:233
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (5) replica removed
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.AmbiguousResultError (4) *errbase.opaqueWrapper (5) *errutil.leafError
    kvnemesis_test.go:70: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestKVNemesisSingleNode3580340144
--- FAIL: TestKVNemesisSingleNode (14.64s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #115598 kv/kvnemesis: TestKVNemesisSingleNode_ReproposalChaos failed [A-kv-transactions C-bug C-test-failure O-robot P-1 T-kv branch-release-23.2]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35872

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -439,6 +439,9 @@ func (v *validator) processOp(txnID *string, op Operation) {
 			// optimistically try to fit the barrier inside one of the current ranges,
 			// but this may race with a split, so we ignore the error in this case and
 			// try again later.
+		} else if resultIsAmbiguous(t.Result) {
+			// It's possible that the replica is removed by the time the barrier
+			// arrives. That's ok, just ignore it.
 		} else {
 			// Fail or retry on other errors, depending on type.
 			v.failIfError(op, t.Result)

